timemachine/fe/single_topology_v3.py

```diff
@@ -269,16 +270,32 @@ def setup_end_state(ff, mol_a, mol_b, core, a_to_c, b_to_c):
     mol_c_torsion_idxs = mol_c_proper_idxs + mol_c_improper_idxs
     mol_c_torsion_params = mol_c_proper_params + mol_c_improper_params
 
-    bond_potential = potentials.HarmonicBond(mol_c_bond_idxs).bind(mol_c_bond_params)
-    angle_potential = potentials.HarmonicAngle(mol_c_angle_idxs).bind(mol_c_angle_params)
-    torsion_potential = potentials.PeriodicTorsion(mol_c_torsion_idxs).bind(mol_c_torsion_params)
+    # canonicalize bonds
+    mol_c_bond_idxs_canon = [canonicalize_bond(idxs) for idxs in mol_c_bond_idxs]
+    bond_potential = potentials.HarmonicBond(mol_c_bond_idxs_canon).bind(mol_c_bond_params)
+
+    mol_c_angle_idxs_canon = [canonicalize_bond(idxs) for idxs in mol_c_angle_idxs]
+    angle_potential = potentials.HarmonicAngle(mol_c_angle_idxs_canon).bind(mol_c_angle_params)
+
+    mol_c_torsion_idxs_canon = [canonicalize_bond(idxs) for idxs in mol_c_torsion_idxs]
+    torsion_potential = potentials.PeriodicTorsion(mol_c_torsion_idxs_canon).bind(mol_c_torsion_params)
 
     # dummy atoms do not have any nonbonded interactions, so we simply turn them off
-    mol_a_nbpl.set_idxs(mol_a_nbpl_idxs)
+
+    mol_c_nbpl_idxs_canon = [canonicalize_bond(idxs) for idxs in mol_a_nbpl_idxs]
+    mol_a_nbpl.set_idxs(mol_c_nbpl_idxs_canon)
     nonbonded_potential = mol_a_nbpl.bind(mol_a_nbpl_params)
 
-    chiral_atom_idxs = np.array(mol_a_chiral_atom_idxs, dtype=np.int32).reshape((-1, 4))
-    chiral_bond_idxs = np.array(mol_a_chiral_bond_idxs, dtype=np.int32).reshape((-1, 4))
+    # chiral atoms need special code for canonicalization, since triple product is invariant
+    # under rotational symmetry (but not something like swap symmetry)
+    canon_chiral_atom_idxs = []
+    for i, j, k, l in mol_a_chiral_atom_idxs:
+        jj, kk, ll = min([(j, k, l), (l, j, k), (k, l, j)])
+        canon_chiral_atom_idxs.append((i, jj, kk, ll))
```

Comment:
Add as a function to dummy?

Reply:
I can't do this on the dummy level - the canonicalization is with respect to the *mapped* idxs in the *combined* molecule, i.e. only after the `recursive_map` has been called

Reply:
Ah, makes sense